spi/README.md

```diff
@@ -112,6 +111,263 @@ Ensure that your **plugin declaration class** implements `ResourceSharingExtensi
 
 ---
 
+### **6. Creating a Client Accessor **
+To ensure a single instance of the `ResourceSharingClient`. CLIENT will be set by security plugin if enabled via `assignResourceSharingClient`. Default to `NoopResourceSharingClient` when CLIENT is null:
+
+```java
+/**
+ * Accessor for resource sharing client supplied by the SPI.
+ */
+public class ResourceSharingClientAccessor {
+  private static ResourceSharingClient CLIENT;
+
+  private ResourceSharingClientAccessor() {}
+
+  /**
+   * Set the resource sharing client
+   */
+  public static void setResourceSharingClient(ResourceSharingClient client) {
+    CLIENT = client;
+  }
+
+  /**
+   * Get the resource sharing client
+   */
+  public static ResourceSharingClient getResourceSharingClient() {
+    return CLIENT == null ? new NoopResourceSharingClient() : CLIENT;
+  }
+}
+```
+
+---
+
+### **7. Using the Client in a Transport Action**
+The following example demonstrates how to use the **Resource Sharing Client** inside a `TransportAction` to verify **delete permissions** before deleting a resource.
+
+```java
+@Inject
+public DeleteResourceTransportAction(TransportService transportService, ActionFilters actionFilters, NodeClient nodeClient) {
+  super(DeleteResourceAction.NAME, transportService, actionFilters, DeleteResourceRequest::new);
+  this.transportService = transportService;
+  this.nodeClient = nodeClient;
+}
+
+@Override
+protected void doExecute(Task task, DeleteResourceRequest request, ActionListener<DeleteResourceResponse> listener) {
+  String resourceId = request.getResourceId();
+  if (resourceId == null || resourceId.isEmpty()) {
+    listener.onFailure(new IllegalArgumentException("Resource ID cannot be null or empty"));
+    return;
+  }
+
+  // Check permission to resource
+  ResourceSharingClient resourceSharingClient = ResourceSharingClientAccessor.getResourceSharingClient();
+  resourceSharingClient.verifyResourceAccess(resourceId, RESOURCE_INDEX_NAME, ActionListener.wrap(isAuthorized -> {
```

Comment:
Do this still make sense to have absent of the `scopes` change that you had earlier?

Authorization will be done as it is currently in the SecurityFilter and based on `cluster_permissions`. In what scenarios would a plugin dev use this method?

Reply:
I'm confused. Are you stating the individual resource access will be evaluated at cluster level?

Reply:
Got it, this fn is used to determine if the resource has been shared w/ the current user.